ashlar/reg.py

```diff
@@ -847,20 +847,49 @@ def _sanitize_channels(self, channels):
             raise ValueError("invalid channels: %s" % invalid_channels)
         return channels
 
+    def _load_single_profile(self, path, num_channels, img_size):
+        """Load, normalize, and validate illumination profile.
+        
+        Parameters
+        ----------
+        path : str
+            Path to the image being loaded.
+        num_channels : int
+            Expected number of channels in the profile image.
+        img_size : tuple
+            shape of a 2D image in (row, column).
+    
+        Returns
+        ----------
+        ndarray
+            Image as numpy array in the (channel, row, column) arrangement. 
+            If ``path`` is ``None``, return a default array of zeros in (channel, 1, 1) shape.
+        """
+        if path is None:
```

Comment:
Suggested refactor:
* Move atleast_3d check below profile.ndim check
* Compute expected_ndim based on num_channels (2 or 3)
* Include given and expected ndim in dimension exception message
* Include given and expected shape in shape exception message (also write "shape" rather than "size")
* Example exception text: `"Illumination profile has dimensionality {profile.ndim}, expected {expected_ndim} to match target image"`